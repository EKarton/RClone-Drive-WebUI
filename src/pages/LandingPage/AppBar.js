import MuiAppBar from '@mui/material/AppBar';
import Button from '@mui/material/Button';
import Toolbar from '@mui/material/Toolbar';
import useScrollTrigger from '@mui/material/useScrollTrigger';
import { Link } from 'react-router-dom';
import AppLogo from 'components/AppLogo';
import DarkModeToggleSwitch from 'components/DarkModeToggleSwitch/index';
import './AppBar.scss';

export default function AppBar() {
  const trigger = useScrollTrigger({
    disableHysteresis: true,
    threshold: 0,
  });

  return (
    <>
      <MuiAppBar
        elevation={trigger ? 1 : 0}
        sx={{ bgcolor: 'background.default', color: 'text.primary' }}
      >
        <Toolbar>
          <AppLogo />
          <div className="landing-page-app-bar__right-items">
            <DarkModeToggleSwitch />
            <Button component={Link} to="https://github.com/EKarton/RClone-Drive-WebUI">
              View Github
            </Button>
            <Button variant="contained" component={Link} to="/login">
              Login
            </Button>
          </div>
        </Toolbar>
      </MuiAppBar>
    </>
  );
}
