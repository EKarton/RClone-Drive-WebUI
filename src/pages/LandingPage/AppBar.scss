.landing-page-app-bar {
  &__right-items {
    display: flex;
    column-gap: 1rem;
  }
}
