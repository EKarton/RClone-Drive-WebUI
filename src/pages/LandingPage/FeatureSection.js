import Typography from '@mui/material/Typography';
import cx from 'classnames';
import './FeatureSection.scss';

export default function FeatureSection({ variant, title, subtitle, imgSrc }) {
  const isTextToRight = variant === 'right';
  const isImageToLeft = !isTextToRight;

  return (
    <div
      className={cx('feature-section', {
        'feature-section--right': isTextToRight,
      })}
    >
      <div className="feature-section__text">
        <Typography variant="h4">
          <strong>{title}</strong>
        </Typography>
        <Typography variant="h6">{subtitle}</Typography>
      </div>
      <img
        src={imgSrc}
        alt="Feature"
        className={cx('feature-section__image', {
          'feature-section__image--left': isImageToLeft,
        })}
      />
    </div>
  );
}
