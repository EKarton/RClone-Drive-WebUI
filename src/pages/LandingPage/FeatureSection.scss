.feature-section {
  display: flex;
  flex-direction: row;
  align-items: center;
  max-height: 80vh;
  margin-left: 6rem;
  margin-right: 0;

  &--right {
    flex-direction: row-reverse;
    margin-left: 0;
    margin-right: 6rem;
  }

  &__text {
    width: 35%;
  }

  &__image {
    height: 80vh;
    object-fit: cover;
    overflow: hidden;
    object-position: right;

    &--left {
      object-position: left;
    }
  }
}

@media (max-width: 900px) {
  .feature-section {
    display: block;
    margin: 3rem 0;

    &__text {
      padding: 2rem;
      width: 100%;
      text-align: center;
    }

    &__image {
      width: 100%;
      height: 100%;
    }
  }
}
