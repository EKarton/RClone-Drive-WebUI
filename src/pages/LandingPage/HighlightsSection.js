import InfoIcon from '@mui/icons-material/Info';
import MouseIcon from '@mui/icons-material/Mouse';
import SecurityIcon from '@mui/icons-material/Security';
import Typography from '@mui/material/Typography';
import './HighlightsSection.scss';

export default function HighlightsSection() {
  return (
    <div className="highlights-section">
      <div className="highlights-section__column">
        <div className="highlights-section__header">
          <InfoIcon color="primary" fontSize="large" />
          <Typography variant="h4">
            <strong>About</strong>
          </Typography>
        </div>
        <Typography variant="h6">
          This is a front-facing web application that connects to your RClone instance.
          You can securely view and manage your files across many cloud providers through
          this web app.
        </Typography>
      </div>
      <div className="highlights-section__column">
        <div className="highlights-section__header">
          <SecurityIcon color="warning" fontSize="large" />
          <Typography variant="h4">
            <strong>Secure</strong>
          </Typography>
        </div>
        <Typography variant="h6">
          All actions are done locally on your computer. Nothing is done in our backend
          servers.
        </Typography>
      </div>
      <div className="highlights-section__column">
        <div className="highlights-section__header">
          <MouseIcon color="success" fontSize="large" />
          <Typography variant="h4">
            <strong>Easy to use</strong>
          </Typography>
        </div>
        <Typography variant="h6">
          Drag-and-drop to upload files, right-click and select to delete, move, or copy
          files between different cloud providers.
        </Typography>
      </div>
    </div>
  );
}
