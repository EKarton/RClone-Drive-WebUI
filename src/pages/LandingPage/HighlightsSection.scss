.highlights-section {
  display: flex;
  justify-content: space-around;
  padding: 6rem;
  gap: 3rem;

  &__column {
    flex-grow: 1;
    flex-basis: 0;
  }

  &__header {
    display: flex;
    justify-items: center;
    column-gap: 0.5rem;
    justify-content: left;
    padding-bottom: 1rem;
  }
}

@media (max-width: 900px) {
  .highlights-section {
    flex-direction: column;
  }
}
