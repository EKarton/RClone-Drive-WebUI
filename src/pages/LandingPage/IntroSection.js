import Button from '@mui/material/Button';
import Typography from '@mui/material/Typography';
import { Link } from 'react-router-dom';
import IntroSectionImage from 'assets/images/landing-page/files-list-page.png';
import './IntroSection.scss';

export default function IntroSection() {
  return (
    <div className="intro-section">
      <div className="intro-section__left-panel">
        <Typography variant="h3" className="intro-section__title">
          <strong>RClone Drive</strong>
        </Typography>
        <Typography variant="h5" className="intro-section__subtitle">
          A RClone client to view and manage your cloud files on the web.
        </Typography>
        <Button variant="contained" component={Link} to={'/login'}>
          Login
        </Button>
      </div>
      <img src={IntroSectionImage} alt="Intro section" className="intro-section__image" />
    </div>
  );
}
