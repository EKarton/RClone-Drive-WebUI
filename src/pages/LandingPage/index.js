import CssBaseline from '@mui/material/CssBaseline';
import Toolbar from '@mui/material/Toolbar';
import DarkModeImage from 'assets/images/landing-page/dark-mode.png';
import FileUploadDialogImage from 'assets/images/landing-page/file-upload-dialog.png';
import ActionsImage from 'assets/images/landing-page/move-file-modal.png';
import FileViewerImage from 'assets/images/landing-page/pdf-viewer.png';
import PicturesListPageImage from 'assets/images/landing-page/pictures-list-page.png';
import AppBar from './AppBar';
import FeatureSection from './FeatureSection';
import HighlightsSection from './HighlightsSection';
import IntroSection from './IntroSection';

const LandingPage = () => {
  return (
    <>
      <CssBaseline />
      <AppBar />
      <Toolbar />
      <IntroSection />
      <HighlightsSection />
      <FeatureSection
        variant="right"
        title="View all pictures"
        subtitle="View and scroll through all of your pictures"
        imgSrc={PicturesListPageImage}
      />
      <FeatureSection
        variant="left"
        title="Open and download files"
        subtitle="View, open, and download PDFs, images, and text documents"
        imgSrc={FileViewerImage}
      />
      <FeatureSection
        variant="right"
        title="Upload new files and folders"
        subtitle="Drag-and-drop to upload files and folder to different cloud drives"
        imgSrc={FileUploadDialogImage}
      />
      <FeatureSection
        variant="left"
        title="Manage files on different cloud drives"
        subtitle="Click to copy, rename, delete, and move files between different cloud drives"
        imgSrc={ActionsImage}
      />
      <FeatureSection
        variant="right"
        title="Dark mode"
        subtitle="Ease your eyes with dark theme"
        imgSrc={DarkModeImage}
      />
    </>
  );
};

export default LandingPage;
