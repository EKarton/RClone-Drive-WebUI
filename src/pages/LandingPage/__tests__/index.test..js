import useScrollTrigger from '@mui/material/useScrollTrigger';
import { customRender } from 'test-utils/react';
import LandingPage from '../index';

jest.mock('@mui/material/useScrollTrigger');

describe('LandingPage', () => {
  beforeEach(() => {
    useScrollTrigger.mockReturnValue(false);
  });

  it('should match snapshot', () => {
    const { baseElement } = customRender(<LandingPage />);

    expect(baseElement).toMatchSnapshot();
  });

  it('should match snapshot when user scrolls down on the page', () => {
    useScrollTrigger.mockReturnValue(true);

    const { baseElement } = customRender(<LandingPage />);

    expect(baseElement).toMatchSnapshot();
  });
});
