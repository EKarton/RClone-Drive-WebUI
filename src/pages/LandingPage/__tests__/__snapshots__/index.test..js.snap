// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage should match snapshot 1`] = `
<body>
  <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-t4svo2-MuiPaper-root-MuiAppBar-root"
      style="--Paper-shadow: none;"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-95f9g5-MuiToolbar-root"
      >
        <div
          class="app-logo"
        />
        <div
          class="landing-page-app-bar__right-items"
        >
          <span
            class="MuiSwitch-root MuiSwitch-sizeSmall css-ds9z5m-MuiSwitch-root"
          >
            <span
              class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-byenzh-MuiButtonBase-root-MuiSwitch-switchBase"
            >
              <input
                class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                type="checkbox"
              />
              <span
                class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
              />
            </span>
            <span
              class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
            />
          </span>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-13n5hkc-MuiTypography-root-MuiLink-root-MuiButtonBase-root-MuiButton-root"
            href="https://github.com/EKarton/RClone-Drive-WebUI"
            tabindex="0"
          >
            View Github
          </a>
          <a
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-142372x-MuiButtonBase-root-MuiButton-root"
            href="/login"
            tabindex="0"
          >
            Login
          </a>
        </div>
      </div>
    </header>
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-95f9g5-MuiToolbar-root"
    />
    <div
      class="intro-section"
    >
      <div
        class="intro-section__left-panel"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 intro-section__title css-27pmhn-MuiTypography-root"
        >
          <strong>
            RClone Drive
          </strong>
        </h3>
        <h5
          class="MuiTypography-root MuiTypography-h5 intro-section__subtitle css-ku7d13-MuiTypography-root"
        >
          A RClone client to view and manage your cloud files on the web.
        </h5>
        <a
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-142372x-MuiButtonBase-root-MuiButton-root"
          href="/login"
          tabindex="0"
        >
          Login
        </a>
      </div>
      <img
        alt="Intro section"
        class="intro-section__image"
        src="files-list-page.png"
      />
    </div>
    <div
      class="highlights-section"
    >
      <div
        class="highlights-section__column"
      >
        <div
          class="highlights-section__header"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeLarge css-rlj5h7-MuiSvgIcon-root"
            data-testid="InfoIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m1 15h-2v-6h2zm0-8h-2V7h2z"
            />
          </svg>
          <h4
            class="MuiTypography-root MuiTypography-h4 css-10zsxzp-MuiTypography-root"
          >
            <strong>
              About
            </strong>
          </h4>
        </div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-8zgow4-MuiTypography-root"
        >
          This is a front-facing web application that connects to your RClone instance. You can securely view and manage your files across many cloud providers through this web app.
        </h6>
      </div>
      <div
        class="highlights-section__column"
      >
        <div
          class="highlights-section__header"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorWarning MuiSvgIcon-fontSizeLarge css-vojx2p-MuiSvgIcon-root"
            data-testid="SecurityIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 1 3 5v6c0 5.55 3.84 10.74 9 12 5.16-1.26 9-6.45 9-12V5zm0 10.99h7c-.53 4.12-3.28 7.79-7 8.94V12H5V6.3l7-3.11z"
            />
          </svg>
          <h4
            class="MuiTypography-root MuiTypography-h4 css-10zsxzp-MuiTypography-root"
          >
            <strong>
              Secure
            </strong>
          </h4>
        </div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-8zgow4-MuiTypography-root"
        >
          All actions are done locally on your computer. Nothing is done in our backend servers.
        </h6>
      </div>
      <div
        class="highlights-section__column"
      >
        <div
          class="highlights-section__header"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorSuccess MuiSvgIcon-fontSizeLarge css-quk7e7-MuiSvgIcon-root"
            data-testid="MouseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M13 1.07V9h7c0-4.08-3.05-7.44-7-7.93M4 15c0 4.42 3.58 8 8 8s8-3.58 8-8v-4H4zm7-13.93C7.05 1.56 4 4.92 4 9h7z"
            />
          </svg>
          <h4
            class="MuiTypography-root MuiTypography-h4 css-10zsxzp-MuiTypography-root"
          >
            <strong>
              Easy to use
            </strong>
          </h4>
        </div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-8zgow4-MuiTypography-root"
        >
          Drag-and-drop to upload files, right-click and select to delete, move, or copy files between different cloud providers.
        </h6>
      </div>
    </div>
    <div
      class="feature-section feature-section--right"
    >
      <div
        class="feature-section__text"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-10zsxzp-MuiTypography-root"
        >
          <strong>
            View all pictures
          </strong>
        </h4>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-8zgow4-MuiTypography-root"
        >
          View and scroll through all of your pictures
        </h6>
      </div>
      <img
        alt="Feature"
        class="feature-section__image"
        src="pictures-list-page.png"
      />
    </div>
    <div
      class="feature-section"
    >
      <div
        class="feature-section__text"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-10zsxzp-MuiTypography-root"
        >
          <strong>
            Open and download files
          </strong>
        </h4>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-8zgow4-MuiTypography-root"
        >
          View, open, and download PDFs, images, and text documents
        </h6>
      </div>
      <img
        alt="Feature"
        class="feature-section__image feature-section__image--left"
        src="pdf-viewer.png"
      />
    </div>
    <div
      class="feature-section feature-section--right"
    >
      <div
        class="feature-section__text"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-10zsxzp-MuiTypography-root"
        >
          <strong>
            Upload new files and folders
          </strong>
        </h4>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-8zgow4-MuiTypography-root"
        >
          Drag-and-drop to upload files and folder to different cloud drives
        </h6>
      </div>
      <img
        alt="Feature"
        class="feature-section__image"
        src="file-upload-dialog.png"
      />
    </div>
    <div
      class="feature-section"
    >
      <div
        class="feature-section__text"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-10zsxzp-MuiTypography-root"
        >
          <strong>
            Manage files on different cloud drives
          </strong>
        </h4>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-8zgow4-MuiTypography-root"
        >
          Click to copy, rename, delete, and move files between different cloud drives
        </h6>
      </div>
      <img
        alt="Feature"
        class="feature-section__image feature-section__image--left"
        src="move-file-modal.png"
      />
    </div>
    <div
      class="feature-section feature-section--right"
    >
      <div
        class="feature-section__text"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-10zsxzp-MuiTypography-root"
        >
          <strong>
            Dark mode
          </strong>
        </h4>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-8zgow4-MuiTypography-root"
        >
          Ease your eyes with dark theme
        </h6>
      </div>
      <img
        alt="Feature"
        class="feature-section__image"
        src="dark-mode.png"
      />
    </div>
  </div>
</body>
`;

exports[`LandingPage should match snapshot when user scrolls down on the page 1`] = `
<body>
  <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation1 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-t4svo2-MuiPaper-root-MuiAppBar-root"
      style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-95f9g5-MuiToolbar-root"
      >
        <div
          class="app-logo"
        />
        <div
          class="landing-page-app-bar__right-items"
        >
          <span
            class="MuiSwitch-root MuiSwitch-sizeSmall css-ds9z5m-MuiSwitch-root"
          >
            <span
              class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-byenzh-MuiButtonBase-root-MuiSwitch-switchBase"
            >
              <input
                class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                type="checkbox"
              />
              <span
                class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
              />
            </span>
            <span
              class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
            />
          </span>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-13n5hkc-MuiTypography-root-MuiLink-root-MuiButtonBase-root-MuiButton-root"
            href="https://github.com/EKarton/RClone-Drive-WebUI"
            tabindex="0"
          >
            View Github
          </a>
          <a
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-142372x-MuiButtonBase-root-MuiButton-root"
            href="/login"
            tabindex="0"
          >
            Login
          </a>
        </div>
      </div>
    </header>
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-95f9g5-MuiToolbar-root"
    />
    <div
      class="intro-section"
    >
      <div
        class="intro-section__left-panel"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 intro-section__title css-27pmhn-MuiTypography-root"
        >
          <strong>
            RClone Drive
          </strong>
        </h3>
        <h5
          class="MuiTypography-root MuiTypography-h5 intro-section__subtitle css-ku7d13-MuiTypography-root"
        >
          A RClone client to view and manage your cloud files on the web.
        </h5>
        <a
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-142372x-MuiButtonBase-root-MuiButton-root"
          href="/login"
          tabindex="0"
        >
          Login
        </a>
      </div>
      <img
        alt="Intro section"
        class="intro-section__image"
        src="files-list-page.png"
      />
    </div>
    <div
      class="highlights-section"
    >
      <div
        class="highlights-section__column"
      >
        <div
          class="highlights-section__header"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeLarge css-rlj5h7-MuiSvgIcon-root"
            data-testid="InfoIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m1 15h-2v-6h2zm0-8h-2V7h2z"
            />
          </svg>
          <h4
            class="MuiTypography-root MuiTypography-h4 css-10zsxzp-MuiTypography-root"
          >
            <strong>
              About
            </strong>
          </h4>
        </div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-8zgow4-MuiTypography-root"
        >
          This is a front-facing web application that connects to your RClone instance. You can securely view and manage your files across many cloud providers through this web app.
        </h6>
      </div>
      <div
        class="highlights-section__column"
      >
        <div
          class="highlights-section__header"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorWarning MuiSvgIcon-fontSizeLarge css-vojx2p-MuiSvgIcon-root"
            data-testid="SecurityIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 1 3 5v6c0 5.55 3.84 10.74 9 12 5.16-1.26 9-6.45 9-12V5zm0 10.99h7c-.53 4.12-3.28 7.79-7 8.94V12H5V6.3l7-3.11z"
            />
          </svg>
          <h4
            class="MuiTypography-root MuiTypography-h4 css-10zsxzp-MuiTypography-root"
          >
            <strong>
              Secure
            </strong>
          </h4>
        </div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-8zgow4-MuiTypography-root"
        >
          All actions are done locally on your computer. Nothing is done in our backend servers.
        </h6>
      </div>
      <div
        class="highlights-section__column"
      >
        <div
          class="highlights-section__header"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorSuccess MuiSvgIcon-fontSizeLarge css-quk7e7-MuiSvgIcon-root"
            data-testid="MouseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M13 1.07V9h7c0-4.08-3.05-7.44-7-7.93M4 15c0 4.42 3.58 8 8 8s8-3.58 8-8v-4H4zm7-13.93C7.05 1.56 4 4.92 4 9h7z"
            />
          </svg>
          <h4
            class="MuiTypography-root MuiTypography-h4 css-10zsxzp-MuiTypography-root"
          >
            <strong>
              Easy to use
            </strong>
          </h4>
        </div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-8zgow4-MuiTypography-root"
        >
          Drag-and-drop to upload files, right-click and select to delete, move, or copy files between different cloud providers.
        </h6>
      </div>
    </div>
    <div
      class="feature-section feature-section--right"
    >
      <div
        class="feature-section__text"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-10zsxzp-MuiTypography-root"
        >
          <strong>
            View all pictures
          </strong>
        </h4>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-8zgow4-MuiTypography-root"
        >
          View and scroll through all of your pictures
        </h6>
      </div>
      <img
        alt="Feature"
        class="feature-section__image"
        src="pictures-list-page.png"
      />
    </div>
    <div
      class="feature-section"
    >
      <div
        class="feature-section__text"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-10zsxzp-MuiTypography-root"
        >
          <strong>
            Open and download files
          </strong>
        </h4>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-8zgow4-MuiTypography-root"
        >
          View, open, and download PDFs, images, and text documents
        </h6>
      </div>
      <img
        alt="Feature"
        class="feature-section__image feature-section__image--left"
        src="pdf-viewer.png"
      />
    </div>
    <div
      class="feature-section feature-section--right"
    >
      <div
        class="feature-section__text"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-10zsxzp-MuiTypography-root"
        >
          <strong>
            Upload new files and folders
          </strong>
        </h4>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-8zgow4-MuiTypography-root"
        >
          Drag-and-drop to upload files and folder to different cloud drives
        </h6>
      </div>
      <img
        alt="Feature"
        class="feature-section__image"
        src="file-upload-dialog.png"
      />
    </div>
    <div
      class="feature-section"
    >
      <div
        class="feature-section__text"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-10zsxzp-MuiTypography-root"
        >
          <strong>
            Manage files on different cloud drives
          </strong>
        </h4>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-8zgow4-MuiTypography-root"
        >
          Click to copy, rename, delete, and move files between different cloud drives
        </h6>
      </div>
      <img
        alt="Feature"
        class="feature-section__image feature-section__image--left"
        src="move-file-modal.png"
      />
    </div>
    <div
      class="feature-section feature-section--right"
    >
      <div
        class="feature-section__text"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-10zsxzp-MuiTypography-root"
        >
          <strong>
            Dark mode
          </strong>
        </h4>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-8zgow4-MuiTypography-root"
        >
          Ease your eyes with dark theme
        </h6>
      </div>
      <img
        alt="Feature"
        class="feature-section__image"
        src="dark-mode.png"
      />
    </div>
  </div>
</body>
`;
